docs/developer/providers.md

```diff
@@ -15,6 +15,189 @@ Currently, Ratify supports a Configuration based Policy Provider named `configPo
 
 The executor is the "glue" that links all Ratify plugin-based components such as the verifiers, referrer stores, and policy providers. The policy provider is primarily invoked for each reference artifact discovered AFTER a configured verifier is found that can be used to verify that particular artifacy type. If a reference artifact is found that does not have a corresponding verifier configured that can perform verification, the executor will ignore this artifact. As a result, this unverifiable artifact will NOT influence the final overall verification success determined by the policy provider.
 
+## Rego Policy Provider
+
+The Rego Policy Provider is a built-in policy provider that uses [Open Policy Agent](https://www.openpolicyagent.org/) (OPA) and [Rego](https://www.openpolicyagent.org/docs/latest/policy-language/) to implement a policy provider. The Rego Policy Provider is a plugin that is registered with Ratify and is invoked by the executor to determine the overall verification success if required. The Rego Policy Provider is configured using the `policy` section of the configuration.
+
+```json
+"policy": {
+    "version": "1.0.0",
+    "plugin": {
+        "name": "regoPolicy",
+        "policyPath": "",
+        "policy": "package ratify.policy\ndefault valid := false\nvalid {\n  not failed_verify(input)\n}\nfailed_verify(reports) {\n  [path, value] := walk(reports)\n  value == false\n  path[count(path) - 1] == \"isSuccess\"\n}"
```

Comment:
we might want to add a note, in the CRD mode, if this policy was edited, any updates will require an ratify pod restart, is that correct?